libmobilecoin/src/chacha20_rng.rs

```diff
@@ -0,0 +1,144 @@
+use crate::common::*;
+use mc_util_ffi::*;
+use rand_chacha::ChaCha20Rng;
+use rand_core::{RngCore, SeedableRng};
+use std::{convert::TryInto, sync::Mutex};
+
+pub type McChaCha20Rng = ChaCha20Rng;
+
+impl_into_ffi!(Mutex<McChaCha20Rng>);
+
+/// Returns a new ChaCha20Rng instance initialized with the
+/// seed value provided by the u64 long_val parameter
+///
+/// # Arguments
+///
+/// * `long_val` - an unsigned 64 bit value to use as the rng seed
+///
+/// # Errors
+///
+/// * `LibMcError::Poison`
+#[no_mangle]
+pub extern "C" fn mc_chacha20_rng_create_with_long(
+    long_val: u64,
+    out_error: FfiOptMutPtr<FfiOptOwnedPtr<McError>>,
+) -> FfiOptOwnedPtr<Mutex<McChaCha20Rng>> {
+    ffi_boundary_with_error(out_error, || {
+        Ok(Mutex::new(McChaCha20Rng::seed_from_u64(long_val)))
+    })
+}
+
+/// Returns a new ChaCha20Rng instance initialized with the
+/// seed value provided by the bytes data, which must be at
+/// least 32 bytes (only the first 32 bytes will be used)
+///
+/// # Arguments
+///
+/// * `bytes` - 32 bytes of data to use as the rng seed
+///
+/// # Errors
+///
+/// * `LibMcError::Poison`
+#[no_mangle]
+pub extern "C" fn mc_chacha20_rng_create_with_bytes(
+    bytes: FfiRefPtr<McBuffer>,
+    out_error: FfiOptMutPtr<FfiOptOwnedPtr<McError>>,
+) -> FfiOptOwnedPtr<Mutex<McChaCha20Rng>> {
+    ffi_boundary_with_error(out_error, || {
+        let bytes: [u8; 32] = bytes
+            .as_slice()
+            .try_into()
+            .expect("seed size must be 32 bytes");
```

Comment:
`bytes.as_slice().try_into()` is very general and spans beyond the case of rngs, so I understand why the expect was used here, but if you'd want to go for a more informative error for this specialized case, it could look something like: 

```suggestion
    ffi_boundary_with_error(out_error, || {
        let bytes: [u8; 32] = bytes
            .as_slice()
            .try_into()
            .map_err(|_| LibMcError::RngInitialization("seed size must be 32 bytes"))?
```